test/System.Net.Http.Formatting.Test/Internal/HttpValueCollectionTest.cs

```diff
@@ -202,29 +205,24 @@ private static void Create_CreateDoesntThrowTooManyValuesPrivate()
         [Fact]
         public void AddTooManyKeysThrows()
         {
-            RunInIsolation(Create_CreateDoesntThrowTooManyValuesPrivate);
+            RunInIsolation(AddTooManyKeysThrowsPrivate);
         }
 
-        private void AddTooManyKeysThrowsPrivate()
+        private static void AddTooManyKeysThrowsPrivate()
         {
             // Note this is static, but also the expected type in a real run.
             MediaTypeFormatter.MaxHttpCollectionKeys = _maxCollectionKeys;
 
             HttpValueCollection collection = CreateInstance();
-
-            int i = 0;
-
-            // Act && Assert
-            Assert.Throws<InvalidOperationException>(() =>
+            for (int i = 0; i < _maxCollectionKeys; i++)
             {
-                for (; i < 1001; i++)
-                {
-                    collection.Add(i.ToString(), i.ToString());
-                }
-            }, TooManyKeysError);
-
+                collection.Add(i.ToString(), i.ToString());
+            }
 
-            Assert.Equal(1000, i);
+            // Act && Assert
+            Assert.Throws<InvalidOperationException>(
+                () => collection.Add(_maxCollectionKeys.ToString(), _maxCollectionKeys.ToString()),
+                TooManyKeysError);
```

Comment:
Turns out the `static` keyword was all I need to get test working everywhere. But, updated test does the same thing and is easier to read.